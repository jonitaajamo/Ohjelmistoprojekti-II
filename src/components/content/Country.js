import React from "react";

let style = {
  opacity: 0.7
};

const Country = ({ country }) => (
  <article className="tile is-child notification is-dark" style={style}>
    <div className="content">
      <p className="title">{country ? country : "Select a country"}</p>
    </div>
  </article>
);

export default Country;
