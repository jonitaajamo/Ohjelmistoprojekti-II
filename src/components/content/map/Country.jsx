import React from "react";

const style = {
  opacity: 0.9,
  margin: "10px",
  position: "absolute",
  top: "0px",
  left: "0px"
};

const Country = ({ country }) => (
  <div className="content" style={style}>
    <span className="tag is-larger is-dark">
      {country ? country : "Select a country"}
    </span>
  </div>
);

export default Country;
