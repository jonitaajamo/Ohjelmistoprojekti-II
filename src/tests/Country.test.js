import React from 'react';
import Country from "../components/content/Country";
import renderer from "react-test-renderer";

it('should display country name', () => {
  const component = renderer.create(
    <Country country='Finland'/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});