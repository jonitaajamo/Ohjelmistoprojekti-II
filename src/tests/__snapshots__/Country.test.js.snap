// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays country name 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Country
    country="Finland"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content"
>
        <p
                className="title"
        >
                Finland
        </p>
</div>,
      "className": "tile is-child notification is-dark",
      "style": Object {
        "opacity": 0.7,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="title"
>
          Finland
</p>,
        "className": "content",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Finland",
          "className": "title",
        },
        "ref": null,
        "rendered": "Finland",
        "type": "p",
      },
      "type": "div",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content"
>
          <p
                    className="title"
          >
                    Finland
          </p>
</div>,
        "className": "tile is-child notification is-dark",
        "style": Object {
          "opacity": 0.7,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="title"
>
            Finland
</p>,
          "className": "content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Finland",
            "className": "title",
          },
          "ref": null,
          "rendered": "Finland",
          "type": "p",
        },
        "type": "div",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
